src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -34,162 +44,541 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+            var dictionaries = ReadDictionaries();
+            if (dictionaries.Any())
+            {
+                var aggregatedDictionary = dictionaries.Aggregate((x, y) => x.CombineWith(y));
+                projectDictionary = projectDictionary.CombineWith(aggregatedDictionary);
+            }
+
+            compilationStartContext.RegisterOperationAction(AnalyzeVariable, OperationKind.VariableDeclarator);
+            compilationStartContext.RegisterCompilationEndAction(AnalyzeAssembly);
+            compilationStartContext.RegisterSymbolAction(
+                AnalyzeSymbol,
+                SymbolKind.Namespace,
+                SymbolKind.NamedType,
+                SymbolKind.Method,
+                SymbolKind.Property,
+                SymbolKind.Event,
+                SymbolKind.Field,
+                SymbolKind.Parameter);
+
+            IEnumerable<CodeAnalysisDictionary> ReadDictionaries()
+            {
+                var fileProvider = AdditionalFileProvider.FromOptions(compilationStartContext.Options);
+                return fileProvider.GetMatchingFiles(@"(?:dictionary|custom).*?\.(?:xml|dic)$")
+                    .Select(CreateDictionaryFromAdditionalText)
+                    .Where(x => x != null);
+
+                CodeAnalysisDictionary CreateDictionaryFromAdditionalText(AdditionalText additionalFile)
+                {
+                    var text = additionalFile.GetText(compilationStartContext.CancellationToken);
+                    var isXml = additionalFile.Path.EndsWith("xml", StringComparison.OrdinalIgnoreCase);
+                    var provider = isXml ? _xmlDictionaryProvider : _dicDictionaryProvider;
+
+                    if (!compilationStartContext.TryGetValue(text, provider, out var dictionary))
+                    {
+                        try
+                        {
+                            // Annoyingly (and expectedly), TryGetValue swallows the parsing exception,
+                            // so we have to parse again to get it.
+                            var unused = isXml ? ParseXmlDictionary(text) : ParseDicDictionary(text);
+                            ReportFileParseDiagnostic(additionalFile.Path, "Unknown error");
+                        }
+#pragma warning disable CA1031 // Do not catch general exception types
+                        catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+                        {
+                            ReportFileParseDiagnostic(additionalFile.Path, ex.Message);
+                        }
+                    }
+
+                    return dictionary;
+                }
+
+                void ReportFileParseDiagnostic(string filePath, string message)
+                {
+                    var diagnostic = Diagnostic.Create(FileParseRule, Location.None, filePath, message);
+                    compilationStartContext.RegisterCompilationEndAction(x => x.ReportDiagnostic(diagnostic));
```

Comment:
❕ No compilation end actions

Reply:
How am I supposed to report diagnostics from CompilationStartActions? Alternatively, which action should I use to do and report on one operation e.g. checking assembly spelling/loading additional files?

Reply:
I think this is fine - assembly level diagnostics will need to be reported at CompilationEndAction or CompilationAction, either one is fine. We have recently added a new API for additional file actions, but that is not yet available, so even those can go in the same compilation level actions.

Reply:
An acceptable alternative would be splitting the analyzer such that the analyzer that requires Compilation End actions is separate. This will allow the normal analyzer to fully function without FSA enabled but still allow the CEA to report assembly diagnostics on build.

Reply:
Such a refactoring is no longer required. The IDE now treats each diagnostic descriptor separately for build-live de-duping, this is not an analyzer level concept anymore. I think a simpler approach would be to create a new diagnostic ID for diagnostics reported at compilation end, and then adding the CompilationEnd custom tag to this descriptor.

Reply:
> each diagnostic descriptor separately

Does this apply for the _descriptor_ or the _ID_?

Reply:
It is at diagnostic ID level, it is not possible to distinguish at descriptor level as diagnostics coming from build only have an associated ID, and we cannot distinguish which action/descriptor maps to the diagnostic.

Reply:
So it sounds like nothing needs to change here, right?

Reply:
Yes, I think you are good here.